kano_video/ui/video.py

```diff
@@ -56,11 +57,14 @@ def __init__(self, e, playlist_name=None, permanent=False):
         self.add(button_grid)
 
         img = Gtk.Image()
-
+        
         if e['thumbnail']:
-            thumbnail = '{}/video_{}.jpg'.format(tmp_dir, time())
-            download_url(e['thumbnail'], thumbnail)
-            img.set_from_file(thumbnail)
+            if os.path.exists(e['thumbnail']):# use local thumbnail if one exists
+                img.set_from_file(e['thumbnail'])
+            else: #download the file if local does not exist
+                thumbnail = '{}/video_{}.jpg'.format(tmp_dir, time())
+                download_url(e['thumbnail'], thumbnail)
+                img.set_from_file(thumbnail)
```

Comment:
Does this actually work? Isn't `e['thumbnail']` a URL but when we download it, the filename is `'video_{}.jpg'.format(time())` which changes every call? As such I think that we will end up downloading the image every time anyways.

If we have a UID for the video then I'm very happy to save the filename including that but the time will constantly change.

Is my thinking correct or am I missing something?